axelor-human-resource/src/main/java/com/axelor/apps/hr/web/timesheet/QuickTimesheetController.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2005-2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.hr.web.timesheet;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.hr.db.Employee;
+import com.axelor.apps.hr.db.Timesheet;
+import com.axelor.apps.hr.db.TimesheetLine;
+import com.axelor.apps.hr.service.timesheet.QuickTimesheetService;
+import com.axelor.apps.hr.service.timesheet.TimesheetFetchService;
+import com.axelor.auth.db.User;
+import com.axelor.auth.db.repo.UserRepository;
+import com.axelor.common.StringUtils;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.axelor.meta.schema.actions.ActionView;
+import com.axelor.rpc.ActionRequest;
+import com.axelor.rpc.ActionResponse;
+import java.math.BigDecimal;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.util.LinkedHashMap;
+
+public class QuickTimesheetController {
+
+  @SuppressWarnings("unchecked")
+  public void computeTimesheet(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+
+    Object userContext = request.getContext().get("user");
+
+    if (userContext != null) {
+
+      User user =
+          userContext instanceof User
+              ? (User) userContext
+              : Beans.get(UserRepository.class)
+                  .find(
+                      Long.valueOf(
+                          ((LinkedHashMap<String, Object>) userContext).get("id").toString()));
+
+      if (user.getEmployee() != null) {
+        LocalDate todayDate = Beans.get(AppBaseService.class).getTodayDate(user.getActiveCompany());
+        LocalDate fromDate = todayDate.with(DayOfWeek.MONDAY);
+        LocalDate toDate = todayDate.with(DayOfWeek.SUNDAY);
+
+        Timesheet timesheet =
+            Beans.get(TimesheetFetchService.class)
+                .getOrCreateOpenTimesheet(user.getEmployee(), fromDate);
```

Comment:
Look to add the toDate in the getOrCreateOpenTimesheet to define timesheet only on the week please